python/ray/data/_internal/plan.py

```diff
@@ -508,20 +484,19 @@ def execute_to_iterator(
     def execute(
         self,
         allow_clear_input_blocks: bool = True,
-        force_read: bool = False,
         preserve_order: bool = False,
-    ) -> BlockList:
+    ) -> RefBundle:
```

Comment:
i think going back to the [discussion](https://github.com/ray-project/ray/pull/45489#discussion_r1635283217) we had in the first PR, i would prefer to have `execute()` return `List[RefBundle]`, where each `RefBundle` corresponds to one output block. This would also make it easy to keep consistent with `execute_to_iterator()`, which can return an Iterator over RefBundles

Let's also let others weigh in about this @raulchen @c21

Reply:
Discussed over Slack -- we'll address this in a follow-up